vpr/src/pack/cluster.cpp

```diff
@@ -2451,14 +2623,36 @@ static t_pack_molecule* get_highest_gain_molecule(t_pb* cur_pb,
         return molecule;
     }
 
+    /*
+     * No suitable molecules were found from the above functions - if
+     * attraction groups were created, explore the attraction groups to see if
+     * any suitable molecules can be found.
+     */
+    int num_pulls = attraction_groups.get_att_group_pulls();
```

Comment:
make pulling from the attraction group lines a helper function

Reply:
actually you can push the "pulls" lines into add_cluster_molecule_candidates_by_att_group